src/main/java/com/zendesk/maxwell/replication/BinlogPosition.java

```diff
@@ -68,8 +81,14 @@ public String getFile() {
 		return file;
 	}
 
-	public Long getHeartbeat() {
-		return heartbeat;
+	public Long getLastHeartbeat() {
+		return lastHeartbeat;
+	}
+
+	public void requireLastHeartbeat() {
+		if (lastHeartbeat == null) {
+			throw new RuntimeException("BinlogPosition.lastHeartbeat must be set");
```

Comment:
If we ever hit this code path it's a programmer error - adding a checked exception would just add noise at the callsite since there's nothing code can actually do to recover. How about an AssertionError, since that's unchecked and less likely to be accidentally caught?